math_opt_benchmark/mst/graph/graph.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 The MathOpt Benchmark Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "math_opt_benchmark/mst/graph/graph.h"
+
+#include <cmath>
+
+#include "absl/random/random.h"
+
+constexpr double kTolerance = 1e-5;
+
+namespace math_opt_benchmark {
+
+void debug_graph(std::vector<std::vector<int>> edges) {
+  for (auto &vector : edges) {
+    std::cout << "[D] Graph: " << absl::StrJoin(vector, ",") << std::endl;
+  }
+}
+
+/**
+ * Construct graph from edges with non-zero solution values
+ * @param problem Graph specification with edges, weights, and number of
+ * vertices
+ * @param solution Solution values from LP solve routine
+ */
+Graph::Graph(const MSTProblem &problem, const MSTSolution &solution) {
+  n_ = problem.n;
+  for (int v1 = 0; v1 < n_; ++v1) {
+    edges_.push_back(std::vector<int>(0));
+  }
+  for (int v1 = 0; v1 < n_; ++v1) {
+    for (const int &v2 : problem.edges[v1]) {
```

Comment:
when the role of v1 and v2 are reversed, are we doing to add edges to the graph twice?